File: libraryCoreJava/src/main/java/com/themone/core/base/impl/BaseApp.java
package com.themone.core.base.impl;

import android.app.Application;

/**
 * @author zhiqiang
 * @date 2019-06-05
 * @desc
 */
public class BaseApp extends Application {

    private static BaseApp instance;


    public static synchronized BaseApp getApplication() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }
}

File: libraryCore/src/main/java/com/themone/core/base/impl/BaseApp.kt
package com.themone.core.base.impl

import android.app.Application
import kotlin.properties.Delegates

/**
 * @author zhiqiang
 * @date 2019-06-05
 * @desc
 */
open class BaseApp : Application() {

    override fun onCreate() {
        super.onCreate()
        application = this
    }

    companion object {

        @get:Synchronized
        var application: BaseApp by Delegates.notNull()
            private set
    }
}
